src/Sylius/Bundle/UserBundle/Form/Type/GenderType.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\UserBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ */
+class GenderType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    {
+        $resolver->setDefaults(array(
+            'choices' => array(
+                0 => '',
+                1 => 'sylius.gender.male',
+                2 => 'sylius.gender.female',
```

Comment:
What are these wide variety of genders?

I think the majority of transexuals would still consider themselves to be male or female and even if they weren't then introducing an extra option or 2 is not prohibited by a char(1) structure. There are 24 other characters afterall and numbers if you prefer. Although I'm not an expert in the sex toys industry, I think you might find that businesses of this kind would perhaps classify their customers with different attributes such as 'sexual orientation' to cover this - something that could easily be extended too.

The upside that you mention is countered by a downside of obscurity of your data. If you want to produce simple reports / export from your database you must perform some kind of mapping on the gender field because no human will know what you mean with a gender value of '0' or '1' whereas they will immediately know what you mean by 'M' or 'F'.

The challenge with building a general purpose application like Sylius is to cover the majority of cases and make it flexible enough to cope with the minorities too. I'm not even sure that the case you mention really exists, but if it does, it's got to be less than 1% of cases and perhaps less than 0.1% - and they can implement this if required.